test/grafter/rdf/sparql_test.clj

```diff
@@ -69,7 +82,26 @@
               "?s ?p ?o ."
               "}")
          (#'sparql/rewrite-values-clauses q4 { [:s :p] [[(URI. "http://s1") (URI. "http://p1")]
-                                                      [(URI. "http://s2") (URI. "http://p2")]]}))))
+                                                      [(URI. "http://s2") (URI. "http://p2")]]})))
+    (is (same-query?
+          (str "SELECT * WHERE {"
+               "VALUES (?s ?p) { (UNDEF UNDEF) }"
+               "?s ?p ?o ."
+               "}")
+          (#'sparql/rewrite-values-clauses q4 { [:s :p] ::sparql/undef})))
+
+    (is (same-query?
+          (str "SELECT * WHERE {"
+               "VALUES (?s ?p) { (UNDEF UNDEF) }"
+               "?s ?p ?o ."
+               "}")
+          (#'sparql/rewrite-values-clauses q4 {[:s :p] [::sparql/undef]})))
```

Comment:
The shape of the data in this test seems wrong to me, as the values need to be a list of `[:s :p]` pairs, I think this test should be `[[::sparql/undef ::sparql/undef]]`

Likewise I think we need a test for `[[::sparql/undef (URI. "http://p1")] [(URI. "http://s1") ::sparql/undef]]`.